negotiation/src/main/java/it/eng/negotiation/service/CallbackHandler.java

```diff
@@ -30,21 +34,77 @@ public class CallbackHandler {
      * @param callbackAddress - the URL to which the response needs to be sent.
      * @param jsonNode        - the JSON content to be sent as the body of the request.
      */
-    public void handleCallbackResponse(String callbackAddress, JsonNode jsonNode) {
-        if (!ObjectUtils.isEmpty(callbackAddress)) {
-            // send response to callback URL
-            okhttp3.RequestBody body = okhttp3.RequestBody.create(jsonNode.toPrettyString(), okhttp3.MediaType.parse(MediaType.APPLICATION_JSON_VALUE));
-            okhttp3.Request request = new Request.Builder()
-                    .url(callbackAddress)
-                    .addHeader("Authorization", okhttp3.Credentials.basic("username", "password"))
-                    .post(body)
-                    .build();
-            log.info("Sending response using callback address: " + callbackAddress);
-            try (okhttp3.Response response = client.executeCall(request)) {
-                log.info("Response received: " + response.body().string());
-            } catch (IOException e) {
-                log.error(e.getLocalizedMessage());
-            }
-        }
-    }
+	public void handleCallbackResponse(String callbackAddress, JsonNode jsonNode) {
+		if(!ObjectUtils.isEmpty(callbackAddress)) {
+			// send response to callback URL
+			RequestBody body = RequestBody.create(jsonNode.toPrettyString(), MediaType.parse("application/json"));
+			Request request = new Request.Builder()
+				      .url(callbackAddress)
+				      .addHeader("Authorization", okhttp3.Credentials.basic("milisav@mail.com", "password"))
```

Comment:
Oki, on it. I will refactor this method to receive authorization string (how this will be created, not relevant for this method) and include comments provided here.